runtime/lib/registry/Registry.cpp

```diff
@@ -0,0 +1,50 @@
+#include <cstdint>
+#include <cstdio>
+#include <pybind11/pybind11.h>
+#include <unordered_map>
+
+#include "Types.h"
+
+namespace py = pybind11;
+
+// From PyBind11's documentation:
+//
+//     Do you have any global variables that are pybind11 objects or invoke pybind11 functions in
+//     either their constructor or destructor? You are generally not allowed to invoke any Python
+//     function in a global static context. We recommend using lazy initialization and then
+//     intentionally leaking at the end of the program.
+//
+// https://pybind11.readthedocs.io/en/stable/advanced/misc.html#common-sources-of-global-interpreter-lock-errors
+std::unordered_map<int64_t, py::function> *references;
```

Comment:
Does it need to be thread-local for the async support or a mutex:
```suggestion
thread_local std::unordered_map<int64_t, py::function> *references;
```

Reply:
No, this has to be global. It has to be protected by a mutex before calling `callbackCall` though! Thanks for the catch!